pkg/controller/events/event_forwarder.go

```diff
@@ -0,0 +1,123 @@
+	if event.Type == "Normal" || !(event.Namespace == "default" || strings.HasPrefix(event.Namespace, "vshn-")) {
+		return ctrl.Result{}, nil
+	}
+
+	// Get the object the event is about
+	obj := &unstructured.Unstructured{}
+	obj.SetGroupVersionKind(schema.FromAPIVersionAndKind(event.InvolvedObject.APIVersion, event.InvolvedObject.Kind))
+	if err := e.Get(ctx, types.NamespacedName{Name: event.InvolvedObject.Name, Namespace: event.InvolvedObject.Namespace}, obj); err != nil {
+		if errors.IsNotFound(err) {
+			// InvolvedObject has already been deleted.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		if errors.IsForbidden(err) {
+			// We don't care for events for objects where the controller doesn't have access to.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request
+		log.Error(err, "Can't read object", "object", obj)
+		return ctrl.Result{}, err
+	}
+	// Check if the object has the annotation to forward events
+	annotationEventForward := obj.GetAnnotations()["vshn.net/forward-events-to"]
+	if annotationEventForward != "" {
+		log.Info("Forwarding event", "event", event.Name, "to", annotationEventForward)
+		forwardingObject := strings.Split(annotationEventForward, "/")
+		apiVersion := forwardingObject[0] + "/" + forwardingObject[1]
+		kind := forwardingObject[2]
+		ns := forwardingObject[3]
+		claimName := forwardingObject[4]
+
+		// Get the claim to forward the event to
+		claim := &unstructured.Unstructured{}
+		claim.SetGroupVersionKind(schema.FromAPIVersionAndKind(apiVersion, kind))
+		if err := e.Get(ctx, types.NamespacedName{Name: claimName, Namespace: ns}, claim); err != nil {
+			return ctrl.Result{}, err
+		}
+		nev := &corev1.Event{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      event.Name,
+				Namespace: ns,
+			},
+			InvolvedObject: corev1.ObjectReference{
+				Kind:            kind,
+				Namespace:       ns,
+				Name:            claimName,
+				APIVersion:      apiVersion,
+				ResourceVersion: claim.GetResourceVersion(),
+				UID:             claim.GetUID(),
+			},
+		}
+
+		// Create or patch the new forwarded event
+		_, err := controllerutil.CreateOrPatch(ctx, e.Client, nev, func() error {
+			nev.Count = event.Count
+			nev.FirstTimestamp = event.FirstTimestamp
+			nev.LastTimestamp = event.LastTimestamp
+			nev.Message = event.Message
+			nev.Reason = event.Reason
+			nev.ReportingController = event.ReportingController
+			nev.ReportingInstance = event.ReportingInstance
+			nev.Type = event.Type
+			nev.Source = event.Source
+			return nil
+		})
+
+		if err != nil {
+			log.Error(err, "Can't create or patch the event")
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (e *EventHandler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Event{}).
```

Comment:
There may be many many events to be reconciled in a cluster. We might one to evaluate if we can reduce the amount of events being processed. Check method WithEventFilter(p predicate.Predicate) *Builder if we can do that.

Reply:
Refactored the code to use the `WithEventFilter` to filter out a bunch of events from the get-go.